packages/block-library/src/tag-cloud/index.php

```diff
@@ -15,9 +15,12 @@
 function render_block_core_tag_cloud( $attributes ) {
 	$args      = array(
 		'echo'       => false,
+		'unit'       => preg_replace( '/^\d*\.?\d*/', '', $attributes['smallestFontSize'] ),
```

Comment:
Hi @amustaque97, thank you for your contribution the things work well from the functional point of view 👍 

I found a security vulnerability in this PR that allows a contributor rule to output any CSS even disallowed rules that we need to fix before merging. So in the end a user may be able to use this to totally change the design of the website even though its role does not allow that, can also output SVG backgrounds, etc.

To reproduce for simple position case do the following steps:
- Create a new post and paste the following string on the code editor `<!-- wp:tag-cloud {"smallestFontSize":"8pt;position:fixed","largestFontSize":"22pt;position:fixed"} /-->`.
- Save the post and verify the position of the tag cloud is fixed while a contributor user should not be able to make elements contain fixed positions.

To fix this we can make the regex stricter e.g: something like `/[0-9.]+[a-zA-Z]+$` that forces a number followed by a set of alphabetic characters that are units but does not allow ";" or ".".

Reply:
@jorgefilipecosta How does this happen when both smallest and largest font size are passed through `floatVal`. 

```php
floatVal( $attributes['smallestFontSize'] )
```

This should obey [the rules of casting floats](https://www.php.net/manual/en/language.types.float.php#language.types.float.casting), namely:

> If the string is [numeric](https://www.php.net/manual/en/language.types.numeric-strings.php) or leading numeric then it will resolve to the corresponding float value, otherwise it is converted to zero (0).

What did I miss?

Reply:
@jorgefilipecosta, I'm glad. thank you for taking a look at my PR and finding issues. I'm happy I learnt something new today. Would you mind taking a look at the fix again. 😅  cc @ciampo 

Thank you.

Reply:
@getdave the issue is in [wp_generate_tag_cloud](https://developer.wordpress.org/reference/functions/wp_generate_tag_cloud/) in this line:
`esc_attr( str_replace( ',', '.', $tag_data['font_size'] ) . $args['unit'] )`

The `unit` that was passed, was the result of the regex handling, making to possible to add more css rules.